app/src/main/java/net/grandcentrix/blutufintegration/data/repo/BluetoothRepository.kt

```diff
@@ -100,10 +99,8 @@ class BluetoothRepository {
         }
     }
 
-    fun deviceSupportsBluetooth(context: Context): Boolean {
-        val bluetoothManager =
-            context.getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager?
-        return bluetoothManager?.adapter != null
+    fun checkPreconditions(): List<MissingPrecondition> {
```

Comment:
This change is a really good step.
Ideally you should have your own `enum` or `sealed` class for a missing precondition. Using the `Blutuf.MissingPrecondition` would expose `Blutuf` to the user of the `BluetoothRepository`. But the idea is that the `BluetoothRepository` hides `Blutuf` from its users.

Reply:
You’re right, I'll fix it.